### PR TITLE
Update Prometheus labels & Grafana dashboard

### PR DESCRIPTION
**Pull request by kruai**
## Overview

This PR is being created to address [RHCLOUD-13815](https://issues.redhat.com/browse/RHCLOUD-13815).
It changes the labels on the Prometheus metrics to use the more specific exception, when available. This will allow our Grafana dashboard to distinguish between validation exceptions, inventory exceptions, and uncaught errors.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by ryandillinfelton — APPROVED** [review]

File: dashboards/grafana-dashboard-insights-inventory-general.configmap.yaml [hunk 1]
```diff
@@ -2217,7 +2217,7 @@ data:
               "link": true,
               "linkTargetBlank": true,
               "linkTooltip": "Kibana",
-              "linkUrl": "${kibana:raw},query:(language:lucene,query:'@log_stream:+%22inventory-mq-*%22+AND+message:+%22Error+adding+host+%22+AND+host.reporter:+%22${__cell_1:raw}%22'),sort:!('@timestamp',desc))",
+              "linkUrl": "${kibana:raw},query:(language:lucene,query:'@log_stream:+%22inventory-mq-*%22+AND+message:+%22Validation+error+while+adding+or+updating+host+%22+AND+host.reporter:+%22${__cell_1:raw}%22'),sort:!('@timestamp',desc))",
```
Comment (tahmidefaz) [inline dashboards/grafana-dashboard-insights-inventory-general.configmap.yaml:2220]: @kruai I am not sure if the part `message: Validation error while adding or updating host` is consistent with all types of ValidationExceptions that we throw. We may need to tweak some logging messages as well. Can you confirm if this is not the case?

Comment (tahmidefaz) [reply dashboards/grafana-dashboard-insights-inventory-general.configmap.yaml:2220]: Sounds good! :+1:

**Review by tahmidefaz — APPROVED** [review]
LGTM! Looks like you have some jenkins issue in the PR checks. Also, we will need to push these changes out to the grafana prod around the same time we do a HBI prod release. :+1:

**Review by thearifismail — APPROVED** [review]
LGTM

**Review by ryandillinfelton — APPROVED** [review]

**Comment by ryandillinfelton** [comment]
I think this has some linting issues now or something

**Comment by kruai** [comment]
IDK why some of them are marked as "failed". When I click the details for each of them, they all appear to have succeeded. I'm going to merge, and if it somehow causes an issue, I can revert.

**Pull request merged**